include/merlin_hashtable.cuh

```diff
@@ -204,6 +220,25 @@ class HashTable {
                  "[HierarchicalKV] `io_by_cpu` should not be true when "
                  "`max_hbm_for_vectors` is not 0!");
     initialized_ = true;
+
+    // Create memory pools.
+    MERLIN_CHECK(options_.device_memory_pool.buffer_size ==
```

Comment:
Well, the pool needs an option struct, and we have two pools. To avoid issues for now, I just require them to be the same. Theoretically, you could use different sizes. But that would require some more complex checking later.